src/ProofOfConcepts/ConnectorTypesGallery/ConnectorTypesListItem.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+
+import {
+  DataList,
+  DataListItem,
+  DataListCell,
+  DataListItemRow,
+  DataListItemCells,
+  Flex,
+  FlexItem,
+  Label,
+  LabelGroup,
+} from '@patternfly/react-core';
+import OutlinedQuestionCircleIcon from '@patternfly/react-icons/dist/esm/icons/outlined-question-circle-icon';
+
+import { useTranslation } from '@rhoas/app-services-ui-components';
+
+export type ConnectorTypesListItemProps = {
+  id: string;
+  labels: string[];
+  title: string;
+  verison: string;
+  numLabels: number;
```

Comment:
Ah, my bad I wasn't clear :smile: 

I was suggesting to maybe save the result of `labels.filter( <etc` in a local variable in this function, something like:

```
const displayLabels = labels.filter(
                                (label) =>
                                  label !== 'source' && label !== 'sink'
                              )
                              .sort((a, b) =>
                                a
                                  .toLocaleLowerCase()
                                  .localeCompare(b.toLocaleLowerCase())
                              )
                              .map((label) => (
                                <Label key={label}>{t(label)}</Label>
                              ))}
```

and pull all this logic out of the JSX, then you can do:

```
<LabelGroup numLabels={displayLabels.length}>
    {/*
source and sink stuff here
*/}
{displayLabels}
```

But maybe I'm overengineering it :smile: 

That being said, it's probably good enough to just pass 10  (or even 5, I think it won't be more than 5) like you were doing and then we don't really need this as a property for the component, good enough to just hard-code it in the JSX.

Reply:
Hey I applied these comments, having <LabelGroup numLabels={displayLabels.length}> actually always leaves you one short since it removes removes the first label, so i just did <LabelGroup numLabels={labels.length}> instead.